### ISSUE TITLE
list[extends class] cannot be assigned to list[class]

### ISSUE THREAD

**Issue by iansan5653** [microsoft/pyright#280]
**Describe the bug**
If I create a type that is a list of objects, ie `typing.List[ExampleClass]`, I cannot put objects in that list that are members of classes that extend `ExampleClass`.

**To Reproduce**
See code. This raises the following:
> Argument of type 'list[ExtendingClass]' cannot be assigned to parameter 'a' of type 'List[ExampleClass]'
>    'ExtendingClass' is incompatible with 'ExampleClass'

**Expected behavior**
The checker should allow objects that inherit from the required class, considering that this is allowed in other circumstances, for example, when calling functions.

**Screenshots or Code**
```py
import typing

class ExampleClass():
    ...

class ExtendingClass(ExampleClass):
    ...

ListOfExampleClass = typing.List[ExampleClass]

def works_with_list_of_example(a: ListOfExampleClass):
    return

my_list = [ExtendingClass()]

works_with_list_of_example(my_list)
```

**VS Code extension or command-line**
Extension version `1.0.68`

**Comment by erictraut** [comment]
This is correct behavior, although it might seem somewhat counterintuitive. If you want to understand the details of why, check out PEP 483 (https://www.python.org/dev/peps/pep-0483/). Here's the summary. Lists are mutable collections, so the type parameter is invariant — it must match exactly. Mutable collection types (like Sequence[T] or Iterable[T]) support covariant type parameters, so derived classes are allowed.

To make your example work, you can simply replace `typing.List` with `typing.Sequence`.
